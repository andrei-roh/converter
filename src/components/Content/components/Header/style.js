import styled from 'styled-components';

export const EmptyBlock = styled.div`
  height: 172px;
  width: 100%;

  @media (min-width: 769px) {
    height: 96px;
  }
`;

export const HeaderBlock = styled.div`
  padding: 12px 0;
  width: 100%;
  background-color: rgba(0, 0, 0, 0.12);
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin-bottom: 10px;
  position: fixed;
  top: 0;
  z-index: 2;
  backdrop-filter: blur(6px);

  @media (min-width: 769px) {
    padding: 20px 0;
    flex-direction: row;
  }
`;

export const IconsBlock = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;
