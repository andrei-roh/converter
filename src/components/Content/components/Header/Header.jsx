import { EmptyBlock, HeaderBlock, EmptyElement } from './style';
import Dark from './components/Dark/Dark';
import EntryField from './components/EntryField/EntryField';

const Header = ({
  mainLabelValue,
  mainFieldValue,
  handleMainFieldChange,
  showDark,
  handleShowDark,
}) => {
  return (
    <>
      <EmptyBlock />
      <HeaderBlock>
        <Dark showDark={showDark} handleShowDark={handleShowDark} />
        <EntryField
          mainLabelValue={mainLabelValue}
          mainFieldValue={mainFieldValue}
          handleMainFieldChange={handleMainFieldChange}
        />
        <EmptyElement />
      </HeaderBlock>
    </>
  );
};

export default Header;
