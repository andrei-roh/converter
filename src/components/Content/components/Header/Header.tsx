import { EmptyBlock, HeaderBlock } from './style';
import LanguageSelector from './components/LanguageSelector/LanguageSelector';
import EntryField from './components/EntryField/EntryField';
import BankSelector from './components/BankSelector/BankSelector';
import Dark from './components/Dark/Dark';
import { IHeader } from '../../../../types';

const Header: React.FC<IHeader> = ({
  mainLabelValue,
  mainFieldValue,
  handleMainFieldChange,
}) => {
  return (
    <>
      <EmptyBlock />
      <HeaderBlock>
        <LanguageSelector />
        <Dark />
        <EntryField
          mainLabelValue={mainLabelValue}
          mainFieldValue={mainFieldValue}
          handleMainFieldChange={handleMainFieldChange}
        />
        <BankSelector />
      </HeaderBlock>
    </>
  );
};

export default Header;
