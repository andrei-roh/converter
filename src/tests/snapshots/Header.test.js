import { shallow } from 'enzyme';
import Header from '../../components/Content/components/Header/Header';
import initialState from '../initialState';

let wrapper;

const { belarusRuble } = initialState;

beforeEach(() => {
  wrapper = shallow(
    <Header
      mainLabelValue={belarusRuble.Cur_OfficialRate}
      mainFieldValue={belarusRuble.Cur_Name}
      handleMainFieldChange={null}
    />
  );
});

describe('<Header /> rendering', () => {
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
