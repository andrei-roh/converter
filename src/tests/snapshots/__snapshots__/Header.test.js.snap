// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> rendering renders correctly 1`] = `ShallowWrapper {}`;
